core/ts/__tests__/MaciState.test.ts

```diff
@@ -744,4 +720,416 @@ describe("MaciState", function () {
+        BigInt(1),
+        BigInt(0),
+        BigInt(pollId),
+      );
+
+      const signature = command.sign(user1Keypair.privKey);
+
+      const ecdhKeypair = new Keypair();
+      const sharedKey = Keypair.genEcdhSharedKey(ecdhKeypair.privKey, coordinatorKeypair.pubKey);
+
+      const message = command.encrypt(signature, sharedKey);
+
+      poll.publishMessage(message, ecdhKeypair.pubKey);
+
+      // mock
+      poll.currentMessageBatchIndex = 0;
+      expect(() => poll.processMessages(pollId)).to.throw(
+        "The current message batch index should not be defined if this is the first batch",
+      );
+      poll.currentMessageBatchIndex = undefined;
+    });
+
+    it("should succeed even if send an invalid message", () => {
+      const command = new PCommand(
+        BigInt(user1StateIndex + 1),
+        user1Keypair.pubKey,
+        BigInt(0),
+        BigInt(1),
+        BigInt(0),
+        BigInt(pollId),
+      );
+
+      const signature = command.sign(user1Keypair.privKey);
+
+      const ecdhKeypair = new Keypair();
+      const sharedKey = Keypair.genEcdhSharedKey(ecdhKeypair.privKey, coordinatorKeypair.pubKey);
+
+      const message = command.encrypt(signature, sharedKey);
+
+      poll.publishMessage(message, ecdhKeypair.pubKey);
+      poll.copyStateFromMaci();
+      expect(() => {
+        poll.processMessage(message, ecdhKeypair.pubKey);
+      }).to.throw("invalid state leaf index");
+
+      expect(() => poll.processMessages(pollId)).to.not.throw;
+    });
+
+    it("should throw when called after all messages have been processed", () => {
+      const command = new PCommand(
+        BigInt(user1StateIndex),
+        user1Keypair.pubKey,
+        BigInt(0),
+        BigInt(1),
+        BigInt(0),
+        BigInt(pollId),
+      );
+
+      const signature = command.sign(user1Keypair.privKey);
+
+      const ecdhKeypair = new Keypair();
+      const sharedKey = Keypair.genEcdhSharedKey(ecdhKeypair.privKey, coordinatorKeypair.pubKey);
+
+      const message = command.encrypt(signature, sharedKey);
+
+      poll.publishMessage(message, ecdhKeypair.pubKey);
+
+      poll.processMessages(pollId);
+
+      expect(() => poll.processMessages(pollId)).to.throw("No more messages to process");
+    });
+  });
+
+  describe("processAllMessages", () => {
+    const maciState = new MaciState(STATE_TREE_DEPTH);
+    const pollId = maciState.deployPoll(
+      BigInt(Math.floor(Date.now() / 1000) + duration),
+      maxValues,
+      treeDepths,
+      messageBatchSize,
+      coordinatorKeypair,
+    );
+
+    const poll = maciState.polls[pollId];
+
+    const user1Keypair = new Keypair();
+    // signup the user
+    const user1StateIndex = maciState.signUp(
+      user1Keypair.pubKey,
+      voiceCreditBalance,
+      BigInt(Math.floor(Date.now() / 1000)),
+    );
+
+    it("it should succeed even if send an invalid message", () => {
+      const command = new PCommand(
+        // only 3 users signed up
```

Comment:
same as above, maybe we just said **// place a wrong user state index here** ?

Reply:
definitely